accounts/models.py

```diff
@@ -1,10 +1,33 @@
 from django.db import models
-from django.contrib.auth.models import User
+from django.contrib.auth.models import AbstractUser
+from django.contrib.auth.admin import UserAdmin
+from django.contrib import admin
+from django.conf import settings
+from django.utils.translation import ugettext_lazy as _
 from edumanage.models import Institution
+import re
 
 
+class User(AbstractUser):
+    class Meta(AbstractUser.Meta):
+        swappable = 'AUTH_USER_MODEL'
+        db_table = 'auth_user';
+        verbose_name = _('user')
+        verbose_name_plural = _('users')
+
+User._meta.get_field('username').max_length = 255
+# try also updating the help_text
+try:
+    help_text = User._meta.get_field('username').help_text._proxy____args[0]
+    if help_text != None:
+	User._meta.get_field('username').help_text = _(re.sub("[0-9]+ characters", "255 characters", help_text))
+except:
+    pass
+
+admin.site.register(User, UserAdmin)
```

Comment:
While trying something irrelevant, namely checking if `django-registration` works with Django 1.7 (not that we expected djnro to work as is in such a setup), we noticed `./manage.py shell` borks due apparently due to this line with an 'AppRegistryNotReady' exception:

```python
Traceback (most recent call last):
  File "./manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/home/safts/virtualenvs/djnro_test/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 385, in execute_from_command_line
    utility.execute()
  File "/home/safts/virtualenvs/djnro_test/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 354, in execute
    django.setup()
  File "/home/safts/virtualenvs/djnro_test/local/lib/python2.7/site-packages/django/__init__.py", line 21, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/home/safts/virtualenvs/djnro_test/local/lib/python2.7/site-packages/django/apps/registry.py", line 108, in populate
    app_config.import_models(all_models)
  File "/home/safts/virtualenvs/djnro_test/local/lib/python2.7/site-packages/django/apps/config.py", line 202, in import_models
    self.models_module = import_module(models_module_name)
  File "/usr/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/srv/djnro/accounts/models.py", line 27, in <module>
    admin.site.register(User, UserAdmin)
  File "/home/safts/virtualenvs/djnro_test/local/lib/python2.7/site-packages/django/contrib/admin/sites.py", line 101, in register
    system_check_errors.extend(admin_class.check(model))
  File "/home/safts/virtualenvs/djnro_test/local/lib/python2.7/site-packages/django/contrib/admin/options.py", line 153, in check
    return cls.checks_class().check(cls, model, **kwargs)
  File "/home/safts/virtualenvs/djnro_test/local/lib/python2.7/site-packages/django/contrib/admin/checks.py", line 494, in check
    errors.extend(self._check_list_filter(cls, model))
  File "/home/safts/virtualenvs/djnro_test/local/lib/python2.7/site-packages/django/contrib/admin/checks.py", line 665, in _check_list_filter
    for index, item in enumerate(cls.list_filter)
  File "/home/safts/virtualenvs/djnro_test/local/lib/python2.7/site-packages/django/contrib/admin/checks.py", line 710, in _check_list_filter_item
    get_fields_from_path(model, field)
  File "/home/safts/virtualenvs/djnro_test/local/lib/python2.7/site-packages/django/contrib/admin/utils.py", line 457, in get_fields_from_path
    fields.append(parent._meta.get_field_by_name(piece)[0])
  File "/home/safts/virtualenvs/djnro_test/local/lib/python2.7/site-packages/django/db/models/options.py", line 416, in get_field_by_name
    cache = self.init_name_map()
  File "/home/safts/virtualenvs/djnro_test/local/lib/python2.7/site-packages/django/db/models/options.py", line 445, in init_name_map
    for f, model in self.get_all_related_m2m_objects_with_model():
  File "/home/safts/virtualenvs/djnro_test/local/lib/python2.7/site-packages/django/db/models/options.py", line 563, in get_all_related_m2m_objects_with_model
    cache = self._fill_related_many_to_many_cache()
  File "/home/safts/virtualenvs/djnro_test/local/lib/python2.7/site-packages/django/db/models/options.py", line 577, in _fill_related_many_to_many_cache
    for klass in self.apps.get_models():
  File "/home/safts/virtualenvs/djnro_test/local/lib/python2.7/site-packages/django/utils/lru_cache.py", line 101, in wrapper
    result = user_function(*args, **kwds)
  File "/home/safts/virtualenvs/djnro_test/local/lib/python2.7/site-packages/django/apps/registry.py", line 168, in get_models
    self.check_models_ready()
  File "/home/safts/virtualenvs/djnro_test/local/lib/python2.7/site-packages/django/apps/registry.py", line 131, in check_models_ready
    raise AppRegistryNotReady("Models aren't loaded yet.")
```

We think it's a good idea anyway to move this to `accounts/admin.py`, with such a patch:
```diff
diff --git a/accounts/admin.py b/accounts/admin.py
index ee85a48..79e7d35 100644
--- a/accounts/admin.py
+++ b/accounts/admin.py
@@ -1,8 +1,10 @@
 from django.contrib import admin
-from accounts.models import UserProfile
+from accounts.models import User, UserProfile
+from django.contrib.auth.admin import UserAdmin
 
 
 class UserPrAdmin(admin.ModelAdmin):
     list_display = ('user', 'institution')
 
 admin.site.register(UserProfile, UserPrAdmin)
+admin.site.register(User, UserAdmin)
diff --git a/accounts/models.py b/accounts/models.py
index ff03608..510e93b 100644
--- a/accounts/models.py
+++ b/accounts/models.py
@@ -24,8 +24,6 @@ try:
 except:
     pass
 
-admin.site.register(User, UserAdmin)
-
 class UserProfile(models.Model):
     user = models.OneToOneField(settings.AUTH_USER_MODEL)
     institution = models.ForeignKey(Institution)
```

Do you agree? If yes, would you mind adding this patch to the PR, or perhaps I can also add this during the merge?